internal/gps/source_manager.go

```diff
@@ -148,7 +148,7 @@ var _ SourceManager = &SourceMgr{}
 // gps's SourceManager is intended to be threadsafe (if it's not, please file a
 // bug!). It should be safe to reuse across concurrent solving runs, even on
 // unrelated projects.
-func NewSourceManager(cachedir string) (*SourceMgr, error) {
+func NewSourceManager(cachedir string, log func(v ...interface{})) (*SourceMgr, error) {
```

Comment:
Looking at this again, the log func type actually made plugging in test/bench log functions much cleaner, but they can be adapted to `log.Logger` if needed.

More importantly, I didn't think through the unexported constructor all the way. It falls short of our needs (or it will in a near future PR) since it can't be accessed by the commands.

Can we start heading in a struct-like direction now? The whole package is still internal, so nothing public is thrashing.